sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -262,7 +262,7 @@ ctes
     ;
 
 namedQuery
-    : name=identifier AS? '(' queryNoWith ')'
+    : name=identifier AS? '(' query ')'
```

Comment:
Currently we only allow a query without `With` used in the CTE's named query. This change loose this limit to allow a query with `With` to be a named query. So we can define CTE in CTE subquery.